packages/fuels-core/src/code_gen/custom_types_gen.rs

```diff
@@ -220,15 +295,56 @@ pub fn expand_internal_enum(name: &str, prop: &Property) -> Result<TokenStream,
                 let selector = (dis, tok);
                 Token::Enum(Box::new(selector))
             }
+
+            pub fn new_from_tokens(tokens: &[Token]) -> Self {
+                if tokens.is_empty() {
+                    panic!("Empty tokens array received in `{}::new_from_tokens`",
+                        #enum_name);
+                }
+                // For some reason sometimes we receive arrays that have multiple elements, with the
+                // first token being a `Token::Enum`. We only consider that `Enum` token in that
+                // case
+                // TODO: figure out what is actually happening and if this is normal
```

Comment:
Can you show me an example that this is reproduced? We should document this if that's the case.

Reply:
It's very hard to reproduce in an example because it can only be "displayed" through a `panic!` print. I will create an issue for this and investigate once it is merged though.